improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+
+        if threshold_attribute and threshold_attribute not in cube_name:
+            self.errors.append(
+                f"Cube name '{cube_name}' is not consistent with "
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}'"
+            )
+
+    def check_cell_methods(self, cube):
+        """Checks cell methods are permitted and correct"""
+        if any([substr in cube.name() for substr in PRECIP_ACCUM_NAMES]):
+            msg = f"Expected sum over time cell method for {cube.name()}"
+            if not cube.cell_methods:
+                self.errors.append(msg)
+            else:
+                found_cm = False
+                for cm in cube.cell_methods:
+                    if (
+                        cm.method == PRECIP_ACCUM_CM.method
+                        and cm.coord_names == PRECIP_ACCUM_CM.coord_names
+                    ):
+                        found_cm = True
+                if not found_cm:
+                    self.errors.append(msg)
+
+        if cube.cell_methods:
+            for cm in cube.cell_methods:
+                if cm.method in COMP_CM_METHODS:
+                    self.methods += f" {cm.method} over {cm.coord_names[0]}"
+                    if self.field_type == self.PROB:
+                        if not cm.comments or cm.comments[0] != f"of {self.diagnostic}":
+                            self.errors.append(
+                                f"Cell method {cm} on probability data should have comment "
+                                f"'of {self.diagnostic}'"
+                            )
+                    # check point and bounds on method coordinate
+                    if "time" in cm.coord_names:
+                        if cube.coord("time").bounds is None:
+                            self.errors.append(
+                                f"Cube of{self.methods} has no time bounds"
+                            )
+
+                elif cm in NONCOMP_CMS or cm.method in NONCOMP_CM_METHODS:
+                    self.errors.append(f"Non-standard cell method {cm}")
+                else:
+                    # flag method which might be invalid, but we can't be sure
+                    self.warnings.append(
+                        f"Unexpected cell method {cm}. Please check the standard to "
+                        "ensure this is valid"
+                    )
+
+    def _check_blend_and_model_attributes(self, attrs):
+        """Interprets attributes for model and blending information
+        and checks for self-consistency"""
+        self.blended = True if BLEND_TITLE_SUBSTR in attrs["title"] else False
+
+        if self.blended:
+            if self.model_id_attr not in attrs:
+                self.errors.append(f"No {self.model_id_attr} on blended file")
+            else:
+                codes = attrs[self.model_id_attr].split(" ")
+                names = [MODEL_NAMES[code] for code in codes]
+                self.model = ", ".join(names)
+
```

Comment:
Suggest returning here, delete the `else` and de-indent the rest of the code in this function. 
Another alternative would be to split this into two functions (one for blended, one for non-blended) and move the `if` out into the calling function.

Reply:
Have returned and de-indented as suggested,